diff --git a/src/ConfigRewriter.php b/src/ConfigRewriter.php
index 9c84791..931d017 100644
--- a/src/ConfigRewriter.php
+++ b/src/ConfigRewriter.php
@@ -177,12 +177,6 @@ class ConfigRewriter implements ConfigRewriterInterface {
    * @throws \Drupal\config_rewrite\Exception\NonexistentInitialConfigException
    */
   public function rewriteConfig($original_config, $rewrite, $config_name, $extensionName) {
-    if (empty($original_config)) {
-      $message = 'Tried to rewrite config @config via @module module without initial config present.';
-      $this->logger->error($message, ['@config' => $config_name, '@module' => $extensionName]);
-      throw new NonexistentInitialConfigException("Tried to rewrite config $config_name by $extensionName module without initial config.");
-    }
-
     if (isset($rewrite['config_rewrite']) && $rewrite['config_rewrite'] == 'replace') {
       $rewritten_config = $rewrite;
     } else {
