diff --git a/modules/filter/filter.module b/modules/filter/filter.module
index 6b911cb8a0..73c521cb08 100644
--- a/modules/filter/filter.module
+++ b/modules/filter/filter.module
@@ -1527,7 +1527,7 @@ function _filter_url($text, $filter) {
     // no replacements are made.
     $open_tag = '';
 
-    for ($i = 0; $i < count($chunks); $i++) {
+    for ($i = 0; $chunks && $i < count($chunks); $i++) {
       if ($chunk_type == 'text') {
         // Only process this text if there are no unclosed $ignore_tags.
         if ($open_tag == '') {
@@ -1557,7 +1557,7 @@ function _filter_url($text, $filter) {
       }
     }
 
-    $text = implode($chunks);
+    $text = $chunks ? implode($chunks) : $text;
     // Revert back to the original comment contents
     _filter_url_escape_comments('', FALSE);
     $text = preg_replace_callback('`<!--(.*?)-->`', '_filter_url_escape_comments', $text);
     
     \n
