diff --git a/modules/activity_send_push/activity_send_push.module b/modules/activity_send_push/activity_send_push.module
index 13ed075..03d2ca4 100755
--- a/modules/activity_send_push/activity_send_push.module
+++ b/modules/activity_send_push/activity_send_push.module
@@ -21,6 +21,8 @@ function activity_send_push_activity_insert(ActivityInterface $activity) {
     // Trigger the create action for entities.
     /* @var $create_action \Drupal\activity_send_push\Plugin\ActivitySend\PushActivitySend */
     $create_action = $activity_send_factory->createInstance('push_activity_send');
-    $create_action->create($activity);
+    // Force the activated to be of type "ActivityInterface".
+    /* @var $activity \Drupal\activity_creator\ActivityInterface */
+    return $create_action->createActivity($activity);
   }
 }
diff --git a/modules/activity_send_push/src/MessageTemplatesOverrides.php b/modules/activity_send_push/src/MessageTemplatesOverrides.php
index 5357167..71fe8a0 100644
--- a/modules/activity_send_push/src/MessageTemplatesOverrides.php
+++ b/modules/activity_send_push/src/MessageTemplatesOverrides.php
@@ -48,6 +48,7 @@ class MessageTemplatesOverrides implements ConfigFactoryOverrideInterface {
       'message.template.create_comment_following_node',
       'message.template.create_content_in_joined_group',
       'message.template.create_private_message',
+      'message.template.create_comment_community_post',
     ];
   }
 
diff --git a/modules/activity_send_push/src/Plugin/ActivitySend/PushActivitySend.php b/modules/activity_send_push/src/Plugin/ActivitySend/PushActivitySend.php
index 26db398..b160412 100644
--- a/modules/activity_send_push/src/Plugin/ActivitySend/PushActivitySend.php
+++ b/modules/activity_send_push/src/Plugin/ActivitySend/PushActivitySend.php
@@ -57,14 +57,6 @@ class PushActivitySend extends ActivitySendBase implements ContainerFactoryPlugi
    * Arguments have been adjusted for backwards compatibility with Open Social.
    */
   public static function create($container_or_activity, array $configuration = [], $plugin_id = NULL, $plugin_definition = NULL) {
-    // Open Social 10 backwards compatibility layer.
-    if ($container_or_activity instanceof ActivityInterface) {
-      if (isset($this)) {
-        $this->process($container_or_activity);
-      }
-      return;
-    }
-
     return new static(
       $configuration,
       $plugin_id,
@@ -75,6 +67,15 @@ class PushActivitySend extends ActivitySendBase implements ContainerFactoryPlugi
     );
   }
 
+  /**
+   * {@inheritdoc}
+   *
+   * Arguments have been adjusted again for backwards compatibility with Open Social.
+   */
+  public function createActivity($container_or_activity, array $configuration = [], $plugin_id = NULL, $plugin_definition = NULL) {
+      $this->process($container_or_activity);
+  }
+
   /**
    * {@inheritdoc}
    */
diff --git a/social_pwa.module b/social_pwa.module
index f61f45f..095d6e1 100644
--- a/social_pwa.module
+++ b/social_pwa.module
@@ -77,6 +77,7 @@ function social_pwa_page_attachments(array &$page) {
     // Get the contents of the Manifest.
     'rel' => 'manifest',
     'href' => '/manifest.json',
+    'crossorigin' => 'use-credentials',
   ];
 
   // META: Attach the meta tags.


